src/router/index.js

```diff
@@ -784,8 +784,18 @@ router.beforeEach(async (to, from, next) => {
     !store.isAuthenticated &&
     !allowedUnauthenticatedRoutes.includes(to.name)
   ) {
-    next({ name: 'SignIn' });
-    return;
+    if (to.fullPath === APP_ROUTES.HOME) {
+      next({ name: 'SignIn' });
+      return;
+    } else if (from.path !== APP_ROUTES.SIGN_IN && to.path !== APP_ROUTES.SSO) {
+      next({ name: 'SignIn', query: { redirect_to: to.fullPath } });
+    } else if (
+      from.path === APP_ROUTES.SIGN_IN &&
+      to.path === APP_ROUTES.SSO &&
+      to.query.redirect_to !== from.query.redirect_to
+    ) {
+      next({ name: 'SSO', query: { redirect_to: from.query.redirect_to } });
+    }
```

Comment:
Similarly here, we can apply the early return pattern for improved readability / maintainability. Without having tested this, something along those lines could work: 

```suggestion
  // If the user attempt to visit the home page, skip the redirect_to query parameter.
  if (to.fullPath === APP_ROUTES.HOME) {
    next({ name: 'SignIn' });
    return;
  }

  // If the user didn't initiate an SSO flow, ensure redirect intent is preserved.
  if (from.path !== APP_ROUTES.SIGN_IN && to.path !== APP_ROUTES.SSO) {
    next({ name: 'SignIn', query: { redirect_to: to.fullPath } });
    return;
  }

  // If the user initiated the SSO flow from the sign-in page, preserve the redirect intent.
  if (
    from.path === APP_ROUTES.SIGN_IN &&
    to.path === APP_ROUTES.SSO &&
    to.query.redirect_to !== from.query.redirect_to
  ) {
    next({ name: 'SSO', query: { redirect_to: from.query.redirect_to } });
    return;
  }
```